client/app/dbaas/logs/detail/streams/logs-streams.service.js

```diff
@@ -0,0 +1,128 @@
+     *
+     * @param {any} serviceName
+     * @returns promise which will be resolve to array of streams
+     * @memberof LogsStreamsService
+     */
+    getStreamDetails (serviceName) {
+        return this.getAllStreams(serviceName)
+            .then(streams => {
+                const promises = streams.map(stream =>
+                    this.getStream(serviceName, stream));
+                return this.$q.all(promises);
+            });
+    }
+
+    /**
+     * returns details of a stream
+     *
+     * @param {any} serviceName
+     * @param {any} streamId
+     * @returns promise which will be resolve to stream object
+     * @memberof LogsStreamsService
+     */
+    getStream (serviceName, streamId) {
+        return this.StreamsApiService.all.get({
+            serviceName,
+            streamId
+        })
+            .$promise
+            .then(stream => stream);
+    }
+
+    /**
+     * returns array of stream id's of logged in user
+     *
+     * @param {any} serviceName
+     * @returns promise which will be resolve to array of stream id's
+     * @memberof LogsStreamsService
+     */
+    getAllStreams (serviceName) {
+        return this.LogsApiService.all.streams({ serviceName })
+            .$promise;
+    }
+
+    /**
+     * returns all notifications configured for a given stream
+     *
+     * @param {any} serviceName
+     * @param {any} streamId
+     * @returns promise which will be resolve to array of notifications
+     * @memberof LogsStreamsService
+     */
+    getNotifications (serviceName, streamId) {
+        return this.StreamsApiService.all.notifications({ serviceName, streamId })
+            .$promise;
+    }
+
+    /**
+     * asynchronously gets notifications of a stream
+     *
+     * @param {any} serviceName
+     * @param {any} stream
+     * @returns stream object after adding notifications
+     * @memberof LogsStreamsService
+     */
+    transformStream (serviceName, stream) {
+        stream.notifications = [];
+        // asynchronously fetch all notification of a stream
+        this.getNotifications(serviceName, stream.streamId)
+            .then(notifications => {
+                stream.notifications = notifications;
+            }).catch(this.ServiceHelper.errorHandler("logs_streams_notifications_get_error"));
+        return stream;
+    }
+
+    /**
+     * returns objecy containing total number of streams and total number of streams used
+     *
+     * @param {any} serviceName
+     * @returns quota object containing V (total number streams) and configured (number of streams used)
+     * @memberof LogsStreamsService
+     */
+    getQuota (serviceName) {
+        return this.AccountingAapiService.all.me({ serviceName }).$promise
+            .then(me => {
+                const quota = {
+                    max: me.total.maxNbStream,
+                    configured: me.total.curNbStream,
+                    currentUsage: me.total.curNbStream * 100 / me.total.maxNbStream
```

Comment:
you should round to 2 digit to make sure we don't end up displaying something like 23.1234234235234234% in the frontend

Reply:
this will be always an integer without decimals. I believe it's not required to round it.

Reply:
Ok :)